src/thirdparty/tests/CMakeLists.txt

```diff
@@ -59,10 +56,15 @@ endif()
 # Smoke test for hdf5 third party library
 #------------------------------------------------------------------------------
 if( HDF5_FOUND)
+
+    set( hdf5_smoke_dependencies hdf5 gtest )
+
+    blt_list_append( TO hdf5_smoke_dependencies ELEMENTS mpi IF ENABLE_MPI)
+
     blt_add_executable(NAME hdf5_smoke_test
                        SOURCES hdf5_smoke.cpp
                        OUTPUT_DIR ${TEST_OUTPUT_DIRECTORY}
-                       DEPENDS_ON hdf5 gtest
+                       DEPENDS_ON ${hdf5_smoke_dependencies}
```

Comment:
I guess you are building with an MPI version of HDF5....

Reply:
... but how did this not break 25 days ago... maybe spack has tests off by default.

Reply:
Hmm...were you using a different version of CMake perhaps? I have run into an issue where different versions of CMake wouldn't propagate dependencies correctly...